src/test/java/com/artipie/asto/RemainingTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.artipie.asto;
+
+import java.nio.ByteBuffer;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link Remaining}.
+ * @since 1.0
+ */
+public final class RemainingTest {
+
+    @Test
+    public void readTwice() throws Exception {
+        final ByteBuffer buf = ByteBuffer.allocate(32);
+        final byte[] array = new byte[]{1, 2, 3, 4};
+        buf.put(array);
+        buf.flip();
+        MatcherAssert.assertThat(
+            "unexpected remaining array", new Remaining(buf).bytes(), Matchers.equalTo(array)
+        );
+        MatcherAssert.assertThat(
+            "failed to read array twice", new Remaining(buf).bytes(), Matchers.equalTo(array)
```

Comment:
@g4s8  Let's use `IsEqual`  instead static matchers. Also, this assertion failure message should start with a capital letter